### PR TITLE
BL-74 feature/as a user I can filter asylum case decisions by calendar year versus fiscal year

### PR DESCRIPTION
**Pull request by attaris978**
## Description

1. Created Ant Design Select component to handle switching between Fiscal Year and Calendar Year date modes

2. Added two styling formats to allow for use of either design document, when that finally gets decided:    a) added .less file styling to circumvent Ant Design default styling. This allows for the Select dropdown menu to appear in 
the design document linked here: [https://github.com/BloomTech-Labs/asylum-rg-docs/blob/main/Designs/UIMockup/UiMockup.pdf](url).    b) Commented out two lines in the component module to revert the styling back to the design document linked here: [https://www.figma.com/file/QbtsElzcyp4XrUCA92W0Im/Untitled](url)    c) Having both options available allows for one-step implementation of whatever style design is chosen (and easy duplication of styling in any future Select components).

3. Created a filterReducer.js file to use redux with the new component. Ultimately, the API will be receiving a filter object which contains filter options from the front-end. This filter option (isFiscalYear) is the first to be implemented, and subsequent filters can now easily be added to the reducer.

4. As a bonus, wrote a random-case generator function and used it to pre-populate a 1000-entry data object which is now in the data file. Just to give our frontend people something to work with to see if their work is actually functional.
    
#### Video Link

https://www.loom.com/share/ccf8b3496a03468aa32882cc3f24134e

## Type of change

- [ ✅ ] New feature (non-breaking change which adds functionality)

## Checklist:

- [ ✅ ] My code follows the style guidelines of this project
- [ ✅ ] I have performed a self-review of my own code
- [ ✅ ] I have removed unnecessary comments/console logs from my code
- [ ✅ ] My changes generate no new warnings
- [ ✅ ] I have checked my code and corrected any misspellings
- [ ✅ ] No duplicate code left within changed files
- [ ✅ ] Size of pull request kept to a minimum
- [ ✅ ] Pull request description clearly describes changes made & motivations for said changes

### CODE REVIEW

**Review by VWashingtonCoder — DISMISSED** [review]
Looks very clean and meets the general front end  design purpose of the ticket. It should allow a very simple to use method of sending the request with the condition of calendar year v. fiscal year.

**Review by SafiaOmar — APPROVED** [review]

**Review by vbrambila2 — APPROVED** [review]
The redux changes look good for scaling and moving forward, great job

**Review by MaxwellChrist — DISMISSED** [review]
Looks like really clean code to me. The reducer is set up nicely and is easy to understand. Keep up the good work!

File: src/components/pages/Table/DateFormatSelect.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+// import '../../../styles/FilterMenuStyles/DateFormatSelect.less';
+//If the rounded Select design as shown at https://github.com/BloomTech-Labs/asylum-rg-docs/blob/main/Designs/UIMockup/UiMockup.pdf is decided upon, uncomment the line above
+const { Option } = Select;
+
+const mapStateToProps = state => ({
+  isFiscalYear: state.filterReducer.isFiscalYear,
+});
+
+const DateFormatSelect = ({ isFiscalYear, setDateFilterFormat }) => (
+  <Select
```
Comment (Ricardorouge) [inline src/components/pages/Table/DateFormatSelect.js:15]: very clean code nice and easy to understand exactly what is happening

**Review by Ricardorouge — DISMISSED** [review]
this was very nicely done, I like that you're using redux to keep state managed I think it is very easy to follow, the addition of the .less file is great makes it very easy to customize in the future and I appreciate all the notes added to the code to help future coders. Excellent work as always !

**Pull request merged**